code/game/gamemodes/dynamic/dynamic_rulesets_roundstart.dm

```diff
@@ -87,6 +87,66 @@
 		LAZYADDASSOC(SSjob.dynamic_forced_occupations, new_malf, "AI")
 	return TRUE
 
+//////////////////////////////////////////////
+//                                          //
+//            INTERNAL AFFAIRS              //
+//                                          //
+//////////////////////////////////////////////
+
+/datum/dynamic_ruleset/roundstart/internal_affairs
+	name = "Internal Affairs"
+	antag_flag = ROLE_INTERNAL_AFFAIRS
+	antag_datum = /datum/antagonist/traitor/internal_affairs
+	protected_roles = list(JOB_CAPTAIN, JOB_HEAD_OF_SECURITY, JOB_WARDEN, JOB_SECURITY_OFFICER, JOB_DETECTIVE, JOB_PRISONER)
+	restricted_roles = list(JOB_AI, JOB_CYBORG)
+	required_candidates = 6
+	weight = 4
+	cost = 8
+	scaling_cost = 2
+	requirements = list(8,8,8,8,8,8,8,8,8,8)
+	antag_cap = list("denominator" = 24, "offset" = 3)
```

Comment:
The offset is very scary to me, and you need a very good reason for this. With this in mind it makes me almost certainly not want to have this on by default, or anything below a certain threat level (and ramping up the cost of this significantly).

Dynamic 2021 was designed to drastically limit the number of roundstart antagonists in exchange for both:

A. Spreading out midrounds across the round
B. Being able to more finely control the number of initial roundstart antagonists as threat level goes up

This looks like this ruleset executing (with the weight, cost, and requirements you gave it is going to be **very** common), you're going to have around 4 antagonists at roundstart every time this ruleset procs, which certainly does not play well with how Dynamic 2021 is balanced..